src/instance/mod.rs

```diff
@@ -58,13 +63,166 @@ impl Instance {
     /// on WebAssembly bytes (represented by the Python bytes type).
     #[new]
     #[allow(clippy::new_ret_no_self)]
-    fn new(object: &PyRawObject, bytes: &PyAny) -> PyResult<()> {
+    fn new(
+        object: &PyRawObject,
+        bytes: &PyAny,
+        imported_functions: &'static PyDict,
```

Comment:
Using `'static` here is cheating, but it is true that `imported_functions` must leave longer than the `instance`. I don't know how to express that in Python…

Reply:
The `Instance.host_function_references` keep a reference to it, so code is valid, but we are lying to Rust.

Reply:
> Using `'static` here is cheating, but it is true that `imported_functions` must leave longer than the `instance`. I don't know how to express that in Python…

if I understand correctly what you meant, in Python is just declaring the name in the parent scope (may use `globals`), so that the variable will outlive the current scope.

Reply:
The dict `imported_functions` must live longer than the instance itself. Usually, Rust is able to infer that based on lifetime, but the macros provided by pyo3 are limited in that sense, and the only I found is to use a `'static` lifetime. That's hacky.